### PR TITLE
feat: add helm plugin scaffolding new layout

### PR DESCRIPTION
**Pull request by camilamacedo86**
**Description of the change:**
This PR adds the helm plugin scaffolding the files with the new layout and using the current sdk helm pkg implementation (image)

- added e2e tests for helm new layout. See:`/test/e2e-helm-new/e2e_suite.go` and `./hack/tests/e2e-helm-new.sh` 
- centralize the method ReplaceInFile in the test/internal/utils.go since it is useful for all tests
- Impl e2e tests in shell for the new layout. See `hack/tests/e2e-helm.sh` 

**Motivation**

SDK is in a process to be integrated with KB which means that its project layouts will be aligned. More info : [Integrating Kubebuilder and Operator SDK](https://github.com/kubernetes-sigs/kubebuilder/blob/master/designs/integrating-kubebuilder-and-osdk.md). 

This PR adds the Helm Plugin which will provide the Helm project in the new layout by the command `operator-sdk init --plugins=`helm.operator-sdk.io/v1`. For a further understanding see the WIP: https://github.com/operator-framework/operator-sdk/pull/329 and the doc [Extensible CLI and Scaffolding Plugins](https://github.com/kubernetes-sigs/kubebuilder/blob/master/designs/extensible-cli-and-scaffolding-plugins-phase-1.md).

**Follow-ups**
- **(Blocked by this one)** - Fix OLM/CSV gen commands for Helm/Ansible operators - Done in https://github.com/operator-framework/operator-sdk/pull/3341
- **(Blocked by this one)** Update docs to use plugins instead of the `operator-sdk new --type and add api` commands - Done in: https://github.com/operator-framework/operator-sdk/pull/3326
- **(Blocked by this one)** Remove `operator-sdk new --type=-helm` https://github.com/operator-framework/operator-sdk/pull/3343
- **(Blocked by this one)** Replace tests done in shell for Golang WIP #3423

### CODE REVIEW

**Comment by camilamacedo86** [comment]
Started in; https://github.com/operator-framework/operator-sdk/pull/3295 ( I close to test the CI in my fork and I could not re-open )

File: hack/tests/e2e-helm-new.sh [hunk 1]
```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env bash
```
Comment (asmacdo) [inline hack/tests/e2e-helm-new.sh:1]: Lets not call this "new", since that will not make any sense very soon. Instead, lets rename the old file to "legacy"

File: .travis.yml [hunk 2]
```diff
@@ -149,6 +149,11 @@ jobs:
       name: Helm on Kubernetes
       script: make test-e2e-helm
 
+    # Build and test helm
+    - <<: *test
+      name: Helm on Kubernetes (new)
+      script: make test-e2e-helm-new
```
Comment (estroz) [inline .travis.yml:155]: To make testing quicker, replace the old `Helm on Kubernetes` with this job instead of adding this new one.

Comment (joelanford) [reply .travis.yml:155]: I suggested to @camilamacedo86 to leave the old legacy e2e in until the old code was removed, but this is a pretty good reason to go ahead and remove it. I think there is _very_ little risk removing it now since we're blocking 1.0 until the rest of the legacy helm scaffolding is removed.

TL;DR: :+1:

File: hack/tests/e2e-helm-new.sh (hunk 1 above)
Comment (camilamacedo86) [reply hack/tests/e2e-helm-new.sh:1]: Hi @asmacdo, 

These tests are just to ensure implementation of the helm plugin. See that idea is to remove all tests shell scripts for a Golang implementation that was already started before. After we merge this one I will push a PR with the Go tests and removing both scripts. So, because of this IMO shows fine these names. However, if you see that it would be a blocker for we move forward then we can do that.

File: hack/tests/e2e-helm-new.sh [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+# If using a kind cluster, load the image into all nodes.
+load_image_if_kind "$DEST_IMAGE"
+
+make install
+make deploy IMG="$DEST_IMAGE"
+OPERATORDIR="$(pwd)"
+
+# kind has an issue with certain image registries (ex. redhat's), so use a
+# different test pod image.
+METRICS_TEST_IMAGE="fedora:latest"
```
Comment (joelanford) [inline hack/tests/e2e-helm-new.sh:143]: Nit and non-blocking since this file is temporary: Can we define `METRICS_TEST_IMAGE` globally at the top of the file so we can reuse it here and when we launch the pod that uses it in the `test_operator` function?

File: hack/tests/e2e-helm-new.sh (hunk 1 above)
Comment (camilamacedo86) [reply hack/tests/e2e-helm-new.sh:1]: See: (Blocked by this one) Replace tests via shell for Golang WIP #3423

File: hack/tests/e2e-helm-new.sh (hunk 3 above)
Comment (camilamacedo86) [reply hack/tests/e2e-helm-new.sh]: done

File: hack/tests/e2e-helm.sh [hunk 4]
```diff
@@ -6,92 +6,84 @@ source hack/lib/test_lib.sh
...
-deploy_operator() {
-    kubectl create -f "$OPERATORDIR/deploy/crds/helm.example.com_nginxes_crd.yaml"
-    kubectl create -f "$OPERATORDIR/deploy/service_account.yaml"
-    kubectl create -f "$OPERATORDIR/deploy/cluster_role.yaml"
-    kubectl create -f "$OPERATORDIR/deploy/cluster_role_binding.yaml"
-    kubectl create -f "$OPERATORDIR/deploy/cluster_operator.yaml"
-    kubectl create namespace ${test_namespace}
-}
+# Namespace used to deploy the helm project
+test_namespace="nginx-operator-system"
```
Comment (joelanford) [inline hack/tests/e2e-helm.sh:17]: `test_namespace` is the namespace in which to create the CR, not the operator. The e2e test ensures that the operator properly handles managing CRs in other namespaces. So this needs to be different than the operator namespace.

Comment (joelanford) [reply hack/tests/e2e-helm.sh:17]: since `make deploy` creates and deploys into `nginx-operator-system`, we need a separate variable (e.g. `operator-namespace`) for this value.

We should use `test_namespace` everywhere we were using it before, and we should use `operator_namespace` whenever we made a kubectl call without an explicit `--namespace` flag before.

File: hack/tests/e2e-helm.sh [hunk 5]
```diff
@@ -6,92 +6,98 @@ source hack/lib/test_lib.sh
...
-    kubectl create -f "$OPERATORDIR/deploy/cluster_role_binding.yaml"
-    kubectl create -f "$OPERATORDIR/deploy/cluster_operator.yaml"
+    make install
+    make deploy IMG="$DEST_IMAGE"
     kubectl create namespace ${test_namespace}
 }
 
 remove_operator() {
+    # delete the clusterrolebinding
+    kubectl delete --ignore-not-found clusterrolebinding nginx-operator-system-metrics-reader --clusterrole=nginx-operator-metrics-reader --serviceaccount=nginx-operator-system:default
```
Comment (joelanford) [inline hack/tests/e2e-helm.sh:27]:
```suggestion
    kubectl delete --ignore-not-found clusterrolebinding nginx-operator-system-metrics-reader
```

File: hack/tests/e2e-helm-new.sh (hunk 1 above)
Comment (camilamacedo86) [reply hack/tests/e2e-helm-new.sh]: Now we have just once which address your concern as well.

File: .travis.yml (hunk 2 above)
Comment (camilamacedo86) [reply .travis.yml]: Done

File: hack/tests/e2e-helm.sh (hunk 4 above)
Comment (camilamacedo86) [reply hack/tests/e2e-helm.sh]: Done 👍
